main.c

```diff
@@ -110,13 +118,17 @@ static void pointer_handle_motion(void *data, struct wl_pointer *wl_pointer,
 			}
 		}
 		break;
-	case WL_POINTER_BUTTON_STATE_PRESSED:
-		seat->has_selection = true;
-		seat->selection.x = min(seat->pressed_x, seat->x);
-		seat->selection.y = min(seat->pressed_y, seat->y);
-		seat->selection.width = abs(seat->x - seat->pressed_x);
-		seat->selection.height = abs(seat->y - seat->pressed_y);
-		break;
+	case WL_POINTER_BUTTON_STATE_PRESSED: {
+			int32_t anchor_x = max(0, seat->anchor_x);
```

Comment:
I'm doing this here and not [l.100](https://github.com/emersion/slurp/pull/35/files#diff-2045016cb90d1e65d71c2407a2570927R100) because we don't want the selection to be cropped if the user goes out of bounds

Reply:
You mean you want to prevent the selection to go out of bounds?

The same needs to be done for the upper bound, I suppose.

Reply:
I want the selection to go out of bound. I just don't want it to be cropped.
Example: I'm doing a 100x100 selection and I want to move it to the 0,0 position. If I go out of bounds atop of the screen by 10px, I don't want it to become a 100x90

Reply:
An image is worth a thousand words:

No `max`:
![output-no-max](https://user-images.githubusercontent.com/1719476/61900297-9e85ec00-af1d-11e9-9335-48b0ceba4ffc.gif)

`max`-ing when setting the `anchor`:
![output-crop](https://user-images.githubusercontent.com/1719476/61900375-c9704000-af1d-11e9-9eaa-bbe459072479.gif)